### DISCUSSION TITLE
How to add `data-turbo-track="reload"` or other options for  stylesheet link

### DISCUSSION THREAD

**Discussion by nyrf** [ElMassimo/vite_ruby#54] [Q&A] [answered]
When use webpacker, i can use `<%= javascript_pack_tag 'application', 'data-turbo-track': 'reload' %>` and `<%= stylesheet_pack_tag 'application', media: 'all', 'data-turbo-track': 'reload' %>`. now, i use `<%= vite_javascript_tag 'application', 'data-turbo-track': 'reload'%>`, it seems that only work for `application.js`

<img width="782" alt="WX20210415-204449@2x" src="https://user-images.githubusercontent.com/299532/114871123-878ed800-9e2b-11eb-90ed-0fc2bf0944f1.png">

**Comment by ElMassimo** [comment]
Hi! Like you mention, at the moment [options are only passed on to the script tag](https://github.com/ElMassimo/vite_ruby/blob/main/vite_rails/lib/vite_rails/tag_helpers.rb#L29-L31).

Will need to take a deeper look at `javascript_include_tag` and `stylesheet_link_tag`, to see if it's reasonable to infer which options can be passed to both the script and the style tag.

The goal would be to do something like:

```ruby
vite_javascript_tag 'application', 'data-turbo-track': 'reload', media: 'all'
```

and have the `media` attribute applied to the style tags, and the `data-turbo-track` attribute to script and style tags.

**Reply by ElMassimo** [reply]
While I figure out a solution, in case this is blocking you, this currently this can be done but it's a bit awkward:

```erb
<%= vite_javascript_tag 'application', 'data-turbo-track': 'reload', skip_style_tags: true %>
<%= stylesheet_link_tag *vite_manifest.resolve_entries('application.js').fetch(:stylesheets), 'data-turbo-track': 'reload' %>
```

**Reply by nyrf** [reply]
Thanks, i have remove `application.css` from `application.js` and rename it to `main.css`, then use `vite_stylesheet_tag` to import `main.css`, the shortcoming is that can't use scss.

```erb
  <%= vite_javascript_tag 'application', 'data-turbo-track': 'reload'%>
  <%= vite_stylesheet_tag 'main', 'data-turbo-track': 'reload'%>
```

**Reply by ElMassimo** [reply]
Currently working on #50, should have a version ready with SCSS support today.

**Reply by nyrf** [reply]
Thanks fro your great work. I've submit a feature request for rails and i found a new PR have be submitted. https://github.com/rails/rails/pull/41927, i don't know if that is useful.

**Comment by ElMassimo** [comment, marked as answer]
Released a [new version of `vite_rails`](https://github.com/ElMassimo/vite_ruby/blob/main/vite_rails/CHANGELOG.md#209-2021-04-15) that will use any additional attributes, so using:

```erb
<%= vite_javascript_tag 'application', 'data-turbo-track': 'reload' %>
```

should now work as expected.

@nyrf You can upgrade by running `bundle update vite_rails`.
